torch/csrc/utils/schema_info.cpp

```diff
@@ -170,6 +170,36 @@ bool SchemaInfo::may_alias(
   }
 }
 
+bool SchemaInfo::may_contain_alias(
+    const c10::SchemaArgument& lhs,
+    const c10::SchemaArgument& rhs,
+    bool bidirectional) {
+  bool basic_check = schema_.may_contain_alias(lhs, rhs) || may_alias(lhs, rhs);
+  if (basic_check) {
+    return true;
+  }
+  if (!alias_maps_current_) {
+    generateAliasMaps();
+  }
+  if (bidirectional) {
+    return mayContainAliasImpl(lhs, rhs) || mayContainAliasImpl(rhs, lhs);
+  } else {
+    return mayContainAliasImpl(lhs, rhs);
+  }
+}
+
+bool SchemaInfo::mayContainAliasImpl(
+    const c10::SchemaArgument& lhs,
+    const c10::SchemaArgument& rhs) {
+  bool types_can_alias = schema_.canAliasTypeSetsAlias(
+      schema_.getAliasTypeSetContainedTypes(schema_.mapTypeToAliasTypeSet(
+          schema_.getCorrectList(lhs.type)[lhs.index].type())),
+      schema_.mapTypeToAliasTypeSet(
+          schema_.getCorrectList(rhs.type)[rhs.index].type()));
```

Comment:
could you add some intermediate variables here to make this easier to read?

Reply:
I know you already documented this below but can you add a quick comment on why this is different from the FunctionSchema implementation?

(you could also just add a comment directing readers to the other documentation, search for `Note [` to see examples of how it's often done in pytorch codebase)